front/src/modules/trainschedule/components/DriverTrainScheduleV2/utils.ts

```diff
@@ -163,8 +163,10 @@ export function getAverageSpeed(posA: number, posB: number, speedList: PositionS
   return Math.round(averageSpeed * 3.6 * 10) / 10;
 }
 
-export const isEco = (simulatedTrain: SimulationResponseSuccess) =>
-  simulatedTrain.base.energy_consumption !== simulatedTrain.final_output.energy_consumption;
+export const isEco = (train: TrainScheduleBase) =>
+  (train.schedule && train.schedule?.length > 1) ||
+  train.margins?.values.some((value) => Number(value.split('%')[0]) > 0) ||
+  train.margins?.values.some((value) => Number(value.split('min')[0]) > 0);
```

Comment:
I think you have some issues here : 
- for the schedules, you lost the fix from @Akctarus with the `.some()`
- when you use margins?.values, probably it will return undefined instead of a boolean and you don't want that
- for values, if you test `value > 0`, you have to iterate twice but you can just test that the value is different than 0% AND 0min/100km, all the other values will work as valid margin.
- the CI failing comes from the fact that isEco is used in 2 other components, you have to change the argument you pass to it in DriverTrainScheduleV2.tsx and for DriverTrainScheduleStopListV2.tsx, as its parent is DriverTrainScheduleV2.tsx, maybe you can change the props simulatedTrain to trainRegime and when passing the props in the parent do trainRegime={baseOrEco === BaseOrEco.eco ? simulatedTrain.final_output : simulatedTrain.base}

```suggestion
export const isEco = (train: TrainScheduleBase) => {
 const { schedule, margins } = train;
 return (schedule && schedule.some((item) => item.arrival !== null))) ||
 (margins && (margins.values.some((value) => value !== '0%' && value !== '0min/100km' ));
}
 
```

Reply:
Done